plan-for-extension-exercise/src/planForExtensionExercise/TotalAreaCalculator.java

```diff
@@ -0,0 +1,25 @@
+package planForExtensionExercise;
+
+public class TotalAreaCalculator extends AreaCalculator {
```

Comment:
Absolutely! And thank you for pointing that out. As you said, the context
here doesn't make that an issue, but I enjoy thinking about those things as
well.

On Thu, Feb 5, 2015 at 3:46 PM, Spencer Robinson <notifications@github.com>
wrote:

> In
> plan-for-extension-exercise/src/planForExtensionExercise/TotalAreaCalculator.java
> <https://github.com/siostechcorp/exercises/pull/4#discussion_r24197829>:
>
> > @@ -0,0 +1,25 @@
> > +package planForExtensionExercise;
> > +
> > +public class TotalAreaCalculator extends AreaCalculator {
>
> Another argument in favor of making TotalAreaCalculator a subclass of
> AreaCalculator is that any code currently relying on AreaCalculator could
> be passed a TotalAreaCalculator instead, changing the behavior without
> changing the code that consumes it.
>
> Of course, in an exercise like this there isn't enough context to make
> such decisions. But it's still fun to discuss.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/siostechcorp/exercises/pull/4/files#r24197829>.
>